### PR TITLE
Better Sort static assert error messages

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -1922,36 +1922,51 @@ See_Also:
...
     /+ Unstable sorting uses the quicksort algorithm, which uses swapAt,
        which either uses swap(...), requiring swappable elements, or just
        swaps using assignment.
        Stable sorting uses TimSort, which needs to copy elements into a buffer,
        requiring assignable elements. +/
 {
     import std.range : assumeSorted;
+    enum swappable = hasSwappableElements!Range;
+    enum assignable = hasAssignableElements!Range;
+    static if(ss == SwapStrategy.unstable) {
```
Comment (dkorpel) [inline std/algorithm/sorting.d:1936]:
```suggestion
    static if (ss == SwapStrategy.unstable) {
```

Comment (burner) [reply std/algorithm/sorting.d]: fixed thanks

File: changelog/sort_assert_messaged.dd [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Better static assert messaged for std.algorithm.sorting.sort
```
Comment (RazvanN7) [inline changelog/sort_assert_messaged.dd:1]:
```suggestion
Better static assert messages for `std.algorithm.sorting.sort`
```

**Review by RazvanN7 — APPROVED** [review]

**Comment by RazvanN7** [comment]
@burner , there seem to be some style failures:

```


std/algorithm/sorting.d(1939:19)[error]: Expected `)` instead of `"unstable, the passed Range '"`
--
  | std/algorithm/sorting.d(1941:68)[warn]: Empty declaration
  | std/algorithm/sorting.d(1946:19)[error]: Expected `)` instead of `" passed Range '"`
  | std/algorithm/sorting.d(1947:62)[warn]: Empty declaration
  | std/algorithm/sorting.d(1951:15)[error]: Expected `)` instead of `Range`
  | std/algorithm/sorting.d(1952:39)[warn]: Empty declaration
  | std/algorithm/sorting.d(1955:15)[error]: Expected `)` instead of `Range`
  | std/algorithm/sorting.d(1955:67)[warn]: Empty declaration
  | std/algorithm/sorting.d(1958:15)[error]: Expected `)` instead of `Range`
  | std/algorithm/sorting.d(1958:66)[warn]: Empty declaration
  | std/algorithm/sorting.d(1964:15)[error]: Expected `)` instead of `" argument 'less' when used with the binaryFun!less template"`
  | std/algorithm/sorting.d(1966:37)[warn]: Empty declaration
  | std/algorithm/sorting.d(1934:10)[warn]: Variable swappable is never used.
  | std/algorithm/sorting.d(1935:10)[warn]: Variable assignable is never used.
```

**Comment by burner** [comment]
I think the "Expected `)` " are because of the new static assert error message feature being used.

The warn about the "Variable swappable is never used." is somewhat misleading. They are used in the static assert.

**Comment by burner** [comment]
not sure how to progress, I don't quite feel like fixing the style checker to get this PR in, and I'm also a big fan of the static assert error message thing.

File: changelog/sort_assert_messaged.dd (hunk 2 above)
Comment (burner) [reply changelog/sort_assert_messaged.dd]: fixed

**Comment by RazvanN7** [comment]
As for the other errors, we can either fix libdparse or just use concatenate expressions.

**Comment by burner** [comment]
changed the static assert messages

**Comment by RazvanN7** [comment]
@burner Thanks! Sorry for not being able to use the static assert with multiple messages. Filed this to libdparse: https://github.com/dlang-community/libdparse/issues/484 . I thought about fixing this myself, however, I am leaning towards finishing integrating dmd as a lib with d-scanner. That will fix the root problem of having duplicate implementations.

**Comment by RazvanN7** [comment]
Looks like it's already been fixed: https://github.com/dlang-community/libdparse/pull/486

**Comment by burner** [comment]
@RazvanN7 thank you for all the work you put into this one.

I'll likely take a look at other template functions of this sort in phobos. Having the static assert messages work the next time will be helpful.

**Comment by WebFreak001** [comment]
D-Scanner published as 0.14.0: https://github.com/dlang-community/D-Scanner/releases/tag/v0.14.0

**Comment by RazvanN7** [comment]
Thanks go to @WebFreak001 for fixing libdparse.

**Pull request merged**